src/final_chain/final_chain.cpp

```diff
@@ -78,9 +78,8 @@ class FinalChainImpl final : public FinalChain {
   future<shared_ptr<FinalizationResult>> finalize(NewBlock new_blk,
                                                   finalize_precommit_ext precommit_ext = {}) override {
     auto p = make_shared<promise<shared_ptr<FinalizationResult>>>();
-    executor_([this, new_blk = move(new_blk), precommit_ext = move(precommit_ext), p]() mutable {
-      p->set_value(finalize_(move(new_blk), precommit_ext));
-    });
+    executor_([this, s = shared_from_this(), new_blk = move(new_blk), precommit_ext = move(precommit_ext),
```

Comment:
so what changed that "s = shared_from_this()" is now required here and before we did not need it ?

Reply:
There was a situations when FinalChain was already destructed when we entering this lambda